src/compact.rs

```diff
@@ -139,12 +138,11 @@ pub trait FindexCompact<
 
         // Get the values stored in the reindexed chains.
         let mut reindexed_chain_values = HashMap::with_capacity(chains_to_reindex.len());
-        for (kwi, chain) in chains_to_reindex {
+        for (kwi, chain) in &chains_to_reindex {
             let blocks = chain
-                .into_iter()
-                .flat_map(|(_, chain_value)| chain_value.into_blocks())
-                .collect::<Vec<_>>();
-            reindexed_chain_values.insert(kwi.clone(), IndexedValue::from_blocks(blocks.iter())?);
+                .iter()
+                .flat_map(|(_, chain_value)| chain_value.as_blocks());
+            reindexed_chain_values.insert(kwi.clone(), IndexedValue::from_blocks(blocks)?);
         }
 
         //
```

Comment:
Unused comment

Reply:
This is the "start" of a comment written by @ThibaudDauce:
```        //
        // Call `list_removed_locations` for all words in one pass instead of
        // calling for one location batch for one word to add noise and prevent
        // the database the size of the chains for each keywords.
        //
```

I do not especially appreciate this style of comment, but we do not have any defined style at Cosmian.

Reply:
Yes sometimes I like to add an empty line before a big comment but not attached to it if you want to remove that.

Reply:
Did not see the following comments, github squished them